optuna/samplers/nsgaii/_sampler.py

```diff
@@ -410,29 +410,46 @@ def after_trial(
 
 
 def _calc_crowding_distance(population: List[FrozenTrial]) -> DefaultDict[int, float]:
+
     manhattan_distances: DefaultDict[int, float] = defaultdict(float)
+    if len(population) == 0:
+        return manhattan_distances
+
     for i in range(len(population[0].values)):
         population.sort(key=lambda x: cast(float, x.values[i]))
 
-        v_min = population[0].values[i]
-        v_max = population[-1].values[i]
-        assert v_min is not None
-        assert v_max is not None
-
-        width = v_max - v_min
-        if width == 0:
+        # If population have the same values[i], ignore that value.
+        if population[0].values[i] == population[-1].values[i]:
             continue
 
-        manhattan_distances[population[0].number] = float("inf")
-        manhattan_distances[population[-1].number] = float("inf")
+        vs = (
+            [-float("inf")]
+            + [cast(List[float], population[j].values)[i] for j in range(len(population))]
+            + [float("inf")]
+        )
+
+        # Smallest finite value.
+        v_min = next(x for x in vs if x != -float("inf"))
 
-        for j in range(1, len(population) - 1):
-            v_high = population[j + 1].values[i]
-            v_low = population[j - 1].values[i]
-            assert v_high is not None
-            assert v_low is not None
+        # Largest finite value.
+        v_max = next(x for x in reversed(vs) if x != float("inf"))
 
-            manhattan_distances[population[j].number] += (v_high - v_low) / width
+        width = v_max - v_min
+        if width <= 0:
+            # width == 0 or width == -inf
+            width = 1.0
```

Comment:
Let me clarify my understanding. This condition is satisfied only when there are only 0 (then, `width==-inf`) or 1 (then, `width==0`) unique finite values. In those cases, only `-inf`, `0`, and `inf` can appear as the differences, so the value of `width` doesn't matter.

Reply:
Yes, you are correct.